src/Compilers/CSharp/Portable/Binder/Semantics/BestTypeInferrer.cs

```diff
@@ -74,7 +63,8 @@ public static TypeSymbol InferBestType(ImmutableArray<BoundExpression> exprs, Co
             // SPEC:    If no such S exists, the expressions have no best common type.
 
             // All non-null types are candidates for best type inference.
-            HashSet<TypeSymbol> candidateTypes = new HashSet<TypeSymbol>();
+            IEqualityComparer<TypeSymbol> comparer = conversions.IncludeNullability ? TypeSymbol.EqualsIncludingNullableComparer : TypeSymbol.EqualsConsiderEverything;
```

Comment:
>EqualsConsiderEverything [](start = 142, length = 24)

Does "Everything" include nullability? If so, this `?:` doesn't do anything. If not, then it isn't "Everything".

Reply:
`Everything` does not include nullability. There is a comment on `TypeCompareKind.ConsiderEverything` to consider renaming.

---
In reply to: [223449926](https://github.com/dotnet/roslyn/pull/29727#discussion_r223449926) [](ancestors = 223449926)

Reply:
I think it would be good to promote that to an issue.

---
In reply to: [223829309](https://github.com/dotnet/roslyn/pull/29727#discussion_r223829309) [](ancestors = 223829309,223449926)